modules/ocf_lb/manifests/init.pp

```diff
@@ -0,0 +1,21 @@
+class ocf_lb (
+  Array $vip_names,
+  String $keepalived_secret_lookup,
+) {
+  $virtual_addresses = $vip_names.map |$vip| {
+    ldap_attr($vip, 'ipHostNumber')
+  }
+  $virtual_addresses_v6 = $vip_names.map |$vip| {
+    regsubst(ldap_attr($vip, 'ipHostNumber'), '^(\d+)\.(\d+)\.(\d+)\.(\d+)$', '2607:f140:8801::1:\4')
```

Comment:
oh my god this was confusing code. Since this is specific to our IP numbering scheme now, it is worth abstracting this into a function called `OcfV6FromV4`. To simplify things, I would further suggest also storing this information in ldap. 

If you decide to stick with this, doesn't hurt to tighten the regex with `\d{1,3}`

Reply:
++ to storing this in LDAP

Reply:
time to figure out ldap-overlay

Reply:
#912